src/number_theory/legendre_symbol/quadratic_char.lean

```diff
@@ -41,7 +53,16 @@ begin
   exact is_square_of_char_two' a,
 end
 
-/-- If the finite field `F` has characteristic `≠ 2`, then it has odd cardinatlity. -/
+/-- If the finite field `F` has characteristic `2`, then it has even cardinality. -/
+lemma even_card_of_char_two (hF : ring_char F = 2) : fintype.card F % 2 = 0 :=
+begin
+  rcases finite_field.card F (ring_char F) with ⟨ n, hp, h ⟩,
+  rw hF at h,
+  rw [h, nat.pow_mod],
+  simp only [nat.bit0_mod_two, zero_pow', ne.def, pnat.ne_zero, not_false_iff, nat.zero_mod],
+end
+
+/-- If the finite field `F` has characteristic `≠ 2`, then it has odd cardinality. -/
 lemma odd_card_of_char_ne_two (hF : ring_char F ≠ 2) : fintype.card F % 2 = 1 :=
 begin
```

Comment:
It should be possible to golf this proof, using `even_card_of_char_two` above. It is the contrapositive + the fact that `% 2` numbers are `= 0` xor `= 1`.

Reply:
OK; I'll look into it. (The second proof was there before the first.)

Reply:
It is not really the contrapositive. I have now changed the first into an equivalence; then the two statements can easily be derived from that.

Reply:
(Or rather, added the `iff` version of the first...)